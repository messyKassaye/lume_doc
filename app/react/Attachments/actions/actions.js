import { actions as formActions } from 'react-redux-form';
import superagent from 'superagent';

import { APIURL } from 'app/config.js';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { updateEntity, selectSingleDocument } from 'app/Library/actions/libraryActions';
import api from 'app/utils/api';
import { RequestParams } from 'app/utils/RequestParams';
import { actions as basicReducerActions } from 'app/BasicReducer';

import * as types from './actionTypes';

export function updateFile(file, entity) {
  return async dispatch => {
    await api.post('files', new RequestParams(file));
    const documents = entity.documents.map(f => {
      if (f._id === file._id) {
        return file;
      }
      return f;
    });
    const updatedEntity = Object.assign(entity, { documents });
    dispatch(basicReducerActions.set('viewer/doc', updatedEntity));
    dispatch(updateEntity(updatedEntity));
    await dispatch(selectSingleDocument(updatedEntity));
    dispatch(notify('File updated', 'success'));
  };
}

export function deleteFile(file, entity) {
  return async dispatch => {
    await api.delete('files', new RequestParams({ _id: file._id }));
    const documents = entity.documents.filter(f => f._id !== file._id);

    const updatedEntity = Object.assign(entity, { documents });
    dispatch(basicReducerActions.set('viewer/doc', updatedEntity));
    dispatch(updateEntity(updatedEntity));
    await dispatch(selectSingleDocument(updatedEntity));
    dispatch(notify('File deleted', 'success'));
  };
}

export function uploadAttachment(entity, file, __reducerKey) {
  return dispatch => {
    dispatch({ type: types.START_UPLOAD_ATTACHMENT, entity });
    superagent
      .post(`${APIURL}files/upload/attachment`)
      .set('Accept', 'application/json')
      .set('X-Requested-With', 'XMLHttpRequest')
      .field('entity', entity)
      .attach('file', file, file.name)
      .on('progress', data => {
        dispatch({ type: types.ATTACHMENT_PROGRESS, entity, progress: Math.floor(data.percent) });
      })
      .on('response', result => {
        dispatch({
          type: types.ATTACHMENT_COMPLETE,
          entity,
          file: JSON.parse(result.text),
          __reducerKey,
        });
      })
      .end();
  };
}

export function uploadAttachmentFromUrl(entity, name, url, __reducerKey) {
  return dispatch => {
    dispatch({ type: types.START_UPLOAD_ATTACHMENT, entity });
    api
      .post('files', new RequestParams({ originalname: name, url, entity, type: 'attachment' }))
      .then(newFile => {
        dispatch({ type: types.ATTACHMENT_COMPLETE, entity, file: newFile.json, __reducerKey });
        dispatch(notify('Attachment uploaded', 'success'));
      });
  };
}

export function renameAttachment(entityId, form, __reducerKey, file) {
  return dispatch =>
    api
      .post(
        'files',
        new RequestParams({
          _id: file._id,
          originalname: file.originalname,
        })
      )
      .then(renamedFile => {
        dispatch({
          type: types.ATTACHMENT_RENAMED,
          entity: entityId,
          file: renamedFile.json,
          __reducerKey,
        });
        dispatch(formActions.reset(form));
        dispatch(notify('Attachment renamed', 'success'));
      });
}

export function deleteAttachment(entitySharedId, attachment, __reducerKey) {
  return async dispatch => {
    await api.delete(
      'files',
      new RequestParams({
        _id: attachment._id,
      })
    );
    dispatch({
      type: types.ATTACHMENT_DELETED,
      entity: entitySharedId,
      file: attachment,
      __reducerKey,
    });

    dispatch(notify('Attachment deleted', 'success'));
  };
}

export function loadForm(form, attachment) {
  return dispatch => {
    dispatch(formActions.reset(form));
    dispatch(formActions.load(form, attachment));
  };
}

export function submitForm(form) {
  return dispatch => {
    dispatch(formActions.submit(form));
  };
}

export function resetForm(form) {
  return dispatch => {
    dispatch(formActions.reset(form));
  };
}
