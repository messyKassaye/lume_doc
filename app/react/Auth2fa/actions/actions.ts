/** @format */
import { Dispatch } from 'redux';

import { User } from 'api/users/usersModel';
import { actions } from 'app/BasicReducer';
import { notificationActions } from 'app/Notifications';
import { RequestParams } from 'app/utils/RequestParams';

import Auth2faAPI from '../Auth2faAPI';

export interface enable2faType {
  type: string;
  key: any;
  value: any;
}

export function enable2fa(): enable2faType {
  return actions.setIn('auth/user', 'using2fa', true);
}

export function reset2fa(user: User) {
  return async (dispatch: Dispatch<{}>) => {
    await Auth2faAPI.reset2fa(new RequestParams({ _id: user._id }));
    dispatch(actions.update('users', { ...user, using2fa: false }));
    dispatch(notificationActions.notify('Deleted successfully.', 'success'));
  };
}
