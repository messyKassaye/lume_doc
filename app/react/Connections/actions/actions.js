import { actions } from 'app/BasicReducer';
import { notificationActions } from 'app/Notifications';
import api from 'app/utils/api';
import debounce from 'app/utils/debounce';
import { RequestParams } from 'app/utils/RequestParams';

import * as types from './actionTypes';
import * as uiActions from './uiActions';

export function immidiateSearch(dispatch, searchTerm, connectionType) {
  dispatch(uiActions.searching());

  const requestParams = new RequestParams({ searchTerm, fields: ['title'] });

  return api.get('search', requestParams).then(response => {
    let results = response.json.rows;
    if (connectionType === 'targetRanged') {
      results = results.filter(r => r.documents.length);
    }
    dispatch(actions.set('connections/searchResults', results));
  });
}

const debouncedSearch = debounce(immidiateSearch, 400);

export function search(searchTerm, connectionType) {
  return dispatch => {
    dispatch(actions.set('connections/searchTerm', searchTerm));
    return debouncedSearch(dispatch, searchTerm, connectionType);
  };
}

export function startNewConnection(connectionType, sourceDocument) {
  return dispatch =>
    immidiateSearch(dispatch, '', connectionType).then(() => {
      dispatch(actions.set('connections/searchTerm', ''));
      dispatch(uiActions.openPanel(connectionType, sourceDocument));
    });
}

export function setRelationType(template) {
  return {
    type: types.SET_RELATION_TYPE,
    template,
  };
}

export function setTargetDocument(id) {
  return {
    type: types.SET_TARGET_DOCUMENT,
    id,
  };
}

export function saveConnection(connection, callback = () => {}) {
  return (dispatch, getState) => {
    dispatch({ type: types.CREATING_CONNECTION });
    if (connection.type !== 'basic') {
      connection.language = getState().locale;
    }

    delete connection.type;

    const sourceRelationship = {
      entity: connection.sourceDocument,
      template: null,
      reference: connection.sourceRange,
      file: connection.sourceFile,
    };

    const targetRelationship = { entity: connection.targetDocument, template: connection.template };
    if (connection.targetRange) {
      Object.assign(targetRelationship, {
        reference: connection.targetRange,
        file: connection.targetFile,
      });
    }

    const apiCall = {
      delete: [],
      save: [[sourceRelationship, targetRelationship]],
    };

    return api.post('relationships/bulk', new RequestParams(apiCall)).then(response => {
      dispatch({ type: types.CONNECTION_CREATED });
      callback(response.json);
      dispatch(notificationActions.notify('saved successfully !', 'success'));
    });
  };
}

export function selectRangedTarget(connection, onRangedConnect) {
  return dispatch => {
    dispatch({ type: types.CREATING_RANGED_CONNECTION });
    onRangedConnect(connection.targetDocument);
  };
}
