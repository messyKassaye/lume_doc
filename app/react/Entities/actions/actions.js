import { actions } from 'app/BasicReducer';
import api from 'app/Entities/EntitiesAPI';
import {
  removeDocument,
  removeDocuments,
  unselectAllDocuments,
  unselectDocument,
} from 'app/Library/actions/libraryActions';
import { notificationActions } from 'app/Notifications';
import { actions as relationshipActions } from 'app/Relationships';
import { RequestParams } from 'app/utils/RequestParams';
import { actions as formActions } from 'react-redux-form';

export function saveEntity(entity) {
  return dispatch =>
    api.save(new RequestParams(entity)).then(response => {
      dispatch(notificationActions.notify('Entity saved', 'success'));
      dispatch(formActions.reset('entityView.entityForm'));
      dispatch(actions.set('entityView/entity', response));
      dispatch(relationshipActions.reloadRelationships(response.sharedId));
    });
}

export function resetForm() {
  return dispatch => dispatch(formActions.reset('entityView.entityForm'));
}

export function deleteEntity(entity) {
  return async dispatch => {
    await api.delete(new RequestParams({ sharedId: entity.sharedId }));
    dispatch(notificationActions.notify('Entity deleted', 'success'));
    dispatch(removeDocument(entity));
    await dispatch(unselectDocument(entity._id));
  };
}

export function deleteEntities(entities) {
  return async dispatch => {
    await api.deleteMultiple(new RequestParams({ sharedIds: entities.map(e => e.sharedId) }));
    dispatch(notificationActions.notify('Deletion success', 'success'));
    await dispatch(unselectAllDocuments());
    dispatch(removeDocuments(entities));
  };
}
