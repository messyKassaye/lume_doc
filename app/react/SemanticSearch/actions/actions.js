import Immutable from 'immutable';
import { actions } from 'app/BasicReducer';
import { RequestParams } from 'app/utils/RequestParams';

import api from '../SemanticSearchAPI';

export function fetchSearches() {
  return dispatch =>
    api.getAllSearches().then(response => {
      dispatch(actions.set('semanticSearch/searches', response));
    });
}

export function setEditSearchEntities(entities = []) {
  return dispatch => {
    dispatch(actions.set('semanticSearch/multiedit', entities));
  };
}

export function editSearchEntities(searchId, args) {
  const requestParams = new RequestParams({ searchId, ...args });
  return dispatch =>
    api.getEntitiesMatchingFilters(requestParams).then(response => {
      dispatch(setEditSearchEntities(response));
    });
}

export function selectSemanticSearchDocument(doc) {
  return dispatch => {
    dispatch(actions.set('semanticSearch/selectedDocument', doc));
    dispatch(actions.set('library.sidepanel.tab', 'semantic-search-results'));
  };
}

export function unselectSemanticSearchDocument() {
  return dispatch => {
    dispatch(actions.set('semanticSearch/selectedDocument', {}));
  };
}

function sanitizeSearchFilters(filters) {
  return Object.keys(filters).reduce((partial, key) => {
    const emptyFilter =
      !filters[key] || (typeof filters[key] === 'object' && !Object.keys(filters[key]).length);
    if (emptyFilter) {
      return partial;
    }
    return { ...partial, [key]: filters[key] };
  }, {});
}

export function submitNewSearch(args) {
  const { searchTerm } = args;
  const query = { ...args, searchTerm: '', filters: sanitizeSearchFilters(args.filters) };

  return dispatch =>
    api.search(new RequestParams({ searchTerm, query })).then(() => dispatch(fetchSearches()));
}

export function showSemanticSearch() {
  return dispatch => {
    dispatch(actions.set('semanticSearch/showSemanticSearchPanel', true));
  };
}

export function hideSemanticSearch() {
  return dispatch => {
    dispatch(actions.set('semanticSearch/showSemanticSearchPanel', false));
  };
}

export function deleteSearch(searchId) {
  return dispatch =>
    api.deleteSearch(new RequestParams({ searchId })).then(() => {
      dispatch(fetchSearches());
    });
}

export function stopSearch(searchId) {
  return dispatch =>
    api.stopSearch(new RequestParams({ searchId })).then(() => {
      dispatch(fetchSearches());
    });
}

export function resumeSearch(searchId) {
  return dispatch =>
    api.resumeSearch(new RequestParams({ searchId })).then(() => {
      dispatch(fetchSearches());
    });
}

export function registerForUpdates() {
  return () => api.registerForUpdates();
}

export function updateSearch(updatedSearch) {
  return dispatch => dispatch(actions.update('semanticSearch/searches', updatedSearch));
}

export function addSearchResults(newDocs) {
  return (dispatch, getState) => {
    const currentSearch = getState().semanticSearch.search;
    const newResults = currentSearch.update('results', existingDocs =>
      newDocs.reduce((updatedDocs, newDoc) => {
        if (!updatedDocs.find(d => newDoc._id === d.get('_id'))) {
          return updatedDocs.push(Immutable.fromJS(newDoc));
        }
        return updatedDocs;
      }, existingDocs)
    );
    dispatch(actions.set('semanticSearch/search', newResults));
  };
}

export function getSearch(searchId, args) {
  const requestParams = new RequestParams({ searchId, ...args });
  return (dispatch, getState) =>
    api.getSearch(requestParams).then(search => {
      dispatch(actions.set('semanticSearch/search', search));
      const selectedDoc = getState().semanticSearch.selectedDocument;
      if (selectedDoc) {
        const updatedDoc = search.results.find(doc => doc.sharedId === selectedDoc.get('sharedId'));
        if (updatedDoc) {
          dispatch(actions.set('semanticSearch/selectedDocument', updatedDoc));
        }
      }
    });
}

export function getMoreSearchResults(searchId, args) {
  const requestParams = new RequestParams({ searchId, ...args });
  return dispatch =>
    api
      .getSearch(requestParams)
      .then(search =>
        dispatch(actions.concatIn('semanticSearch/search', ['results'], search.results))
      );
}
