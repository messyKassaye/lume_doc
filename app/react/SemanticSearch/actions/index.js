import {
  fetchSearches,
  setEditSearchEntities,
  editSearchEntities,
  selectSemanticSearchDocument,
  unselectSemanticSearchDocument,
  submitNewSearch,
  showSemanticSearch,
  hideSemanticSearch,
  deleteSearch,
  stopSearch,
  resumeSearch,
  registerForUpdates,
  updateSearch,
  addSearchResults,
  getSearch,
  getMoreSearchResults,
} from './actions';

export default {
  fetchSearches,
  setEditSearchEntities,
  editSearchEntities,
  selectSemanticSearchDocument,
  unselectSemanticSearchDocument,
  submitNewSearch,
  showSemanticSearch,
  hideSemanticSearch,
  deleteSearch,
  stopSearch,
  resumeSearch,
  registerForUpdates,
  updateSearch,
  addSearchResults,
  getSearch,
  getMoreSearchResults,
};
