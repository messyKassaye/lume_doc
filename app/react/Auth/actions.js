/** @format */

import api from 'app/utils/api';
import { actions } from 'app/BasicReducer';
import { notify } from 'app/Notifications/actions/notificationsActions';
import { RequestParams } from 'app/utils/RequestParams';

export function login(credentials) {
  const request = new RequestParams(credentials);
  return async dispatch => {
    await api.post('login', request);
    const user = await api.get('user');
    dispatch(actions.set('auth/user', user.json));
  };
}

export function recoverPassword(email) {
  const request = new RequestParams({ email });
  return dispatch =>
    api.post('recoverpassword', request).then(() => {
      dispatch(
        notify(
          'Instructions to reset your password have been send, please check your email',
          'success'
        )
      );
    });
}

export function resetPassword(password, key) {
  const request = new RequestParams({ password, key });
  return dispatch =>
    api.post('resetpassword', request).then(() => {
      dispatch(notify('Password changed success', 'success'));
    });
}

export function unlockAccount(credentials) {
  const request = new RequestParams(credentials);
  return dispatch =>
    api.post('unlockaccount', request).then(() => {
      dispatch(notify('Account unlocked successfully', 'success'));
    });
}
