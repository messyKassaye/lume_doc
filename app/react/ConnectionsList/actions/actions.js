import { actions } from 'app/BasicReducer';
import { actions as formActions } from 'react-redux-form';
import { notificationActions } from 'app/Notifications';
import referencesAPI from 'app/Viewer/referencesAPI';
import { fromJS as Immutable } from 'immutable';
import prioritySortingCriteria from 'app/utils/prioritySortingCriteria';
import { RequestParams } from 'app/utils/RequestParams';

import * as uiActions from 'app/Entities/actions/uiActions';

export function search(requestParams) {
  const { sharedId, sort, filters } = requestParams.data;
  const searchTerm =
    requestParams.data.search && requestParams.data.search.searchTerm
      ? requestParams.data.search.searchTerm.value
      : '';

  let options = { sharedId, ...sort };
  if (filters) {
    options = { sharedId, ...sort, ...filters.toJS(), searchTerm };
  }
  return referencesAPI.search(requestParams.onlyHeaders().add(options));
}

export function searchReferences() {
  return async (dispatch, getState) => {
    const relationshipsList = getState().relationships.list;
    const results = await search(new RequestParams(relationshipsList));
    dispatch(actions.set('relationships/list/searchResults', results));
    dispatch(uiActions.showTab('connections'));
  };
}

export function connectionsChanged() {
  return (dispatch, getState) => {
    const relationshipsList = getState().relationships.list;
    const { sharedId } = relationshipsList;

    return referencesAPI
      .getGroupedByConnection(new RequestParams({ sharedId }))
      .then(connectionsGroups => {
        const filteredTemplates = connectionsGroups.reduce(
          (templateIds, group) => templateIds.concat(group.templates.map(t => t._id.toString())),
          []
        );

        const sortOptions = prioritySortingCriteria.get({
          currentCriteria: relationshipsList.sort,
          filteredTemplates,
          templates: getState().templates,
        });
        return Promise.all([connectionsGroups, sortOptions]);
      })
      .then(([connectionsGroups, sort]) => {
        dispatch(actions.set('relationships/list/connectionsGroups', connectionsGroups));
        dispatch(formActions.merge('relationships/list.sort', sort));
        return searchReferences()(dispatch, getState);
      });
  };
}

export function deleteConnection(connection) {
  return async (dispatch, getState) => {
    await referencesAPI.delete(new RequestParams({ _id: connection._id }));
    dispatch(notificationActions.notify('Connection deleted', 'success'));
    return connectionsChanged()(dispatch, getState);
  };
}

export function loadAllReferences() {
  return async (dispatch, getState) => {
    const relationshipsList = getState().relationships.list;
    dispatch(
      actions.set('relationships/list/filters', relationshipsList.filters.set('limit', 9999))
    );
    return searchReferences()(dispatch, getState);
  };
}

export function loadMoreReferences(limit) {
  return function(dispatch, getState) {
    const relationshipsList = getState().relationships.list;
    dispatch(
      actions.set('relationships/list/filters', relationshipsList.filters.set('limit', limit))
    );
    return searchReferences()(dispatch, getState);
  };
}

export function setFilter(groupFilterValues) {
  return function(dispatch, getState) {
    const relationshipsList = getState().relationships.list;
    const currentFilter = relationshipsList.filters.get('filter') || Immutable({});
    const newFilter = currentFilter.merge(groupFilterValues);
    dispatch(
      actions.set('relationships/list/filters', relationshipsList.filters.set('filter', newFilter))
    );
    return searchReferences()(dispatch, getState);
  };
}

export function resetSearch() {
  return function(dispatch, getState) {
    dispatch(formActions.change('relationships/list/search.searchTerm', ''));
    dispatch(actions.set('relationships/list/filters', Immutable({})));
    return searchReferences()(dispatch, getState);
  };
}

export function switchView(type) {
  return function(dispatch, getState) {
    dispatch(actions.set('relationships/list/view', type));
    if (type === 'graph') {
      return loadAllReferences()(dispatch, getState);
    }
  };
}
