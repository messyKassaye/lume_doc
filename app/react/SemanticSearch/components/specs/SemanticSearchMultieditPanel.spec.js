/** @format */

import React from 'react';
import Immutable from 'immutable';
import { shallow } from 'enzyme';

import multiReducer from 'app/Multireducer';
import * as metadataActions from 'app/Metadata/actions/actions';
import * as searchActions from 'app/SemanticSearch/actions/actions';
import {
  mapStateToProps,
  mapDispatchToProps,
  SemanticSearchMultieditPanel,
} from '../SemanticSearchMultieditPanel';

describe('SemanticSearchMultieditPanel', () => {
  let state;
  let dispatch;
  let props;

  const mockAction = (obj, fn) => jest.spyOn(obj, fn).mockReturnValue(() => {});
  beforeEach(() => {
    props = {
      storeKey: 'library',
      formKey: 'semanticSearch.multipleEdit',
      searchId: 'searchId',
    };
    state = {
      semanticSearch: {
        multiedit: Immutable.fromJS([
          { searchId: 'doc1', template: 'tpl1' },
          { searchId: 'doc2', template: 'tpl1' },
          { searchId: 'doc3', template: 'tpl2' },
        ]),
        multipleEditForm: {
          metadata: {
            unchangedField: { pristine: true },
            changedField: { pristine: false },
          },
        },
      },
      templates: Immutable.fromJS([
        {
          _id: 'tpl1',
          properties: [{ name: 'p1', type: 'select', content: 't1' }],
        },
        {
          _id: 'tpl2',
          properties: [
            { name: 'p1', type: 'select', content: 't1' },
            { name: 'p2', type: 'text' },
          ],
        },
      ]),
      thesauris: Immutable.fromJS([
        { _id: 't1', name: 'T1', values: [{ _id: 'v1', id: 'v1', label: 'V1' }] },
      ]),
    };
    mockAction(metadataActions, 'loadTemplate');
    mockAction(metadataActions, 'resetReduxForm');
    mockAction(metadataActions, 'multipleUpdate');
    mockAction(searchActions, 'setEditSearchEntities');
    mockAction(searchActions, 'getSearch');
    dispatch = jest.fn().mockImplementation(() => Promise.resolve());
    jest.spyOn(multiReducer, 'wrapDispatch').mockReturnValue(dispatch);
  });

  const getProps = () => ({
    ...props,
    ...mapStateToProps(state),
    ...mapDispatchToProps(dispatch, props),
  });

  const render = () => shallow(<SemanticSearchMultieditPanel {...getProps()} />);

  it('should render multi edit form for semantic search multi edit documents', () => {
    const component = render();
    expect(component).toMatchSnapshot();
  });

  describe('changeTemplate', () => {
    it('should set the template of the entities to the selected one', () => {
      const component = render();
      component.instance().changeTemplate({}, 'tpl3');
      expect(searchActions.setEditSearchEntities.mock.calls).toMatchSnapshot();
    });
  });

  describe('save', () => {
    let formValues;
    let instance;
    let component;
    beforeEach(() => {
      formValues = {
        metadata: { unchangedField: [{ value: 'val1' }], changedField: [{ value: 'val2' }] },
      };
      component = render();
      instance = component.instance();
      spyOn(instance, 'close');
    });

    it('should apply changes to entities and re-fetch the search', async () => {
      await instance.save(formValues);
      expect(metadataActions.multipleUpdate).toHaveBeenCalledWith(state.semanticSearch.multiedit, {
        metadata: { changedField: [{ value: 'val2' }] },
      });
      expect(searchActions.getSearch).toHaveBeenCalledWith('searchId');
      expect(instance.close).toHaveBeenCalled();
    });
    it('should update entities icon if icon changed', async () => {
      formValues.icon = 'icon';
      state.semanticSearch.multipleEditForm.icon = {
        pristine: false,
      };
      await instance.save(formValues);
      expect(metadataActions.multipleUpdate).toHaveBeenCalledWith(state.semanticSearch.multiedit, {
        metadata: { changedField: [{ value: 'val2' }] },
        icon: 'icon',
      });
    });
    it('should set template if common template found', async () => {
      state = {
        ...state,
      };
      state.semanticSearch.multiedit = state.semanticSearch.multiedit.map(item =>
        item.set('template', 'tpl1')
      );
      instance = render().instance();
      spyOn(instance, 'close');
      await instance.save(formValues);
      expect(metadataActions.multipleUpdate).toHaveBeenCalledWith(state.semanticSearch.multiedit, {
        metadata: { changedField: [{ value: 'val2' }] },
        template: 'tpl1',
      });
    });
  });

  describe('close', () => {
    it('should reset form and set entities to an empty list', () => {
      const component = render();
      const instance = component.instance();
      instance.close();
      expect(metadataActions.resetReduxForm).toHaveBeenCalledWith(props.formKey);
      expect(searchActions.setEditSearchEntities).toHaveBeenCalledWith([]);
    });
  });

  describe('open', () => {
    it('should not open side panel if there are no multi edit entities', () => {
      state.semanticSearch.multiedit = Immutable.fromJS([]);
      const component = render();
      expect(component).toMatchSnapshot();
    });
  });
});
