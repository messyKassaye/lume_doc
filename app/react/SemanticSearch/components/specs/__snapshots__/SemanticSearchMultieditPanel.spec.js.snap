// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SemanticSearchMultieditPanel changeTemplate should set the template of the entities to the selected one 1`] = `
Array [
  Array [
    Immutable.List [
      Immutable.Map {
        "searchId": "doc1",
        "template": "tpl3",
      },
      Immutable.Map {
        "searchId": "doc2",
        "template": "tpl3",
      },
      Immutable.Map {
        "searchId": "doc3",
        "template": "tpl3",
      },
    ],
  ],
]
`;

exports[`SemanticSearchMultieditPanel open should not open side panel if there are no multi edit entities 1`] = `
<SidePanel
  className="multi-edit"
  mode=""
  open={false}
>
  <div
    className="sidepanel-header"
  >
    <button
      className="closeSidepanel close-modal"
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
    </button>
  </div>
  <div
    className="sidepanel-body"
  >
    <div
      className="alert alert-warning"
    >
      <Connect(Icon)
        icon="exclamation-triangle"
        size="2x"
      />
      <p>
        Warning: you are editing multiple files. Fields marked with a
         
        <Connect(Icon)
          icon="exclamation-triangle"
        />
         will be updated with the same value.
      </p>
    </div>
    <Connect(MetadataForm)
      changeTemplate={[Function]}
      id="multiEdit"
      model="semanticSearch.multipleEdit"
      multipleEdition={true}
      onSubmit={[Function]}
      template={null}
      thesauris={
        Immutable.List [
          Immutable.Map {
            "_id": "t1",
            "name": "T1",
            "values": Immutable.List [
              Immutable.Map {
                "_id": "v1",
                "id": "v1",
                "label": "V1",
              },
            ],
          },
        ]
      }
    />
  </div>
  <div
    className="sidepanel-footer"
  >
    <button
      className="cancel-edit-metadata btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
      <span
        className="btn-label"
      >
        Cancel
      </span>
    </button>
    <button
      className="btn btn-success"
      form="multiEdit"
      type="submit"
    >
      <Connect(Icon)
        icon="save"
      />
      <span
        className="btn-label"
      >
        Save
      </span>
    </button>
  </div>
</SidePanel>
`;

exports[`SemanticSearchMultieditPanel should render multi edit form for semantic search multi edit documents 1`] = `
<SidePanel
  className="multi-edit"
  mode=""
  open={true}
>
  <div
    className="sidepanel-header"
  >
    <button
      className="closeSidepanel close-modal"
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
    </button>
  </div>
  <div
    className="sidepanel-body"
  >
    <div
      className="alert alert-warning"
    >
      <Connect(Icon)
        icon="exclamation-triangle"
        size="2x"
      />
      <p>
        Warning: you are editing multiple files. Fields marked with a
         
        <Connect(Icon)
          icon="exclamation-triangle"
        />
         will be updated with the same value.
      </p>
    </div>
    <Connect(MetadataForm)
      changeTemplate={[Function]}
      id="multiEdit"
      model="semanticSearch.multipleEdit"
      multipleEdition={true}
      onSubmit={[Function]}
      template={
        Immutable.Map {
          "_id": "",
          "properties": Immutable.List [
            Immutable.Map {
              "name": "p1",
              "type": "select",
              "content": "t1",
            },
          ],
        }
      }
      thesauris={
        Immutable.List [
          Immutable.Map {
            "_id": "t1",
            "name": "T1",
            "values": Immutable.List [
              Immutable.Map {
                "_id": "v1",
                "id": "v1",
                "label": "V1",
              },
            ],
          },
        ]
      }
    />
  </div>
  <div
    className="sidepanel-footer"
  >
    <button
      className="cancel-edit-metadata btn btn-primary"
      onClick={[Function]}
      type="button"
    >
      <Connect(Icon)
        icon="times"
      />
      <span
        className="btn-label"
      >
        Cancel
      </span>
    </button>
    <button
      className="btn btn-success"
      form="multiEdit"
      type="submit"
    >
      <Connect(Icon)
        icon="save"
      />
      <span
        className="btn-label"
      >
        Save
      </span>
    </button>
  </div>
</SidePanel>
`;
